pythonFiles/vscode_datascience_helpers/dataframes/vscodeDataFrame.py

```diff
@@ -105,6 +105,8 @@ def _VSCODE_convertToDataFrame(df, start=None, end=None):
         df = _VSCODE_convertTensorToDataFrame(df, start, end)
     elif hasattr(vartype, "__name__") and vartype.__name__ == "ndarray":
         df = _VSCODE_convertNumpyArrayToDataFrame(df, start, end)
+    elif hasattr(df, "__array__") and hasattr(vartype, "__name__") and vartype.__name__ == "DataArray":
+        df = _VSCODE_convertNumpyArrayToDataFrame(df.__array__(), start, end)
```

Comment:
This might be slow, especially when opening the data viewer from the debug variables context menu, since the debug adapter rate-limits how many rows we can fetch at a time (we set this limit to 100 but it really depends on the number of columns in the data as well). This would result in us converting the entire xarray but only returning the [start:end] rows in the request. Suggestion: try opening some 2D and higher dimensional xarrays in our data viewer with the current implementation, see if it's significantly slower. It may be that converting only the rows we need makes this faster (I don't know whether that's the case).

Reply:
I see what you're saying, and I saw you mention that doing `df[start:end].__array__()` might be better, which makes sense to me. It was definitely faster on (1000, 1000, 1000) xarrays.